Backend: migrate Spring AI to 2.0.0-M5 (unblocks dependabot #70)

Dependabot #70 bumps `org.springframework.ai:spring-ai-bom` from 2.0.0-M4 → 2.0.0-M5 but CI fails:

```
[ERROR] components/promptlm-execution-springai/src/main/java/dev/promptlm/execution/openai/OpenAiVendorClient.java:[30,41]
        package org.springframework.ai.openai.api does not exist
```

Spring AI reorganized the package layout in M5. The bump can't merge until consumers are migrated.

## Scope

- Update imports / API usage in `components/promptlm-execution-springai/` (start with `OpenAiVendorClient.java`).
- Verify other Spring AI consumers compile (`promptlm-execution`, anywhere else `org.springframework.ai.*` is referenced).
- Rebase #70 on the migration commit.

## Notes

- This is **non-blocking** for the v1 critical path — the existing M4 build works fine.
- Do this work alongside #70 (or close #70 and re-open after migration; dependabot will recreate).
- Test with `mvn -pl components/promptlm-execution-springai test` before pushing.

Refs: #70

--- a/components/promptlm-execution-springai/src/main/java/dev/promptlm/execution/openai/OpenAiVendorClient.java
+++ b/components/promptlm-execution-springai/src/main/java/dev/promptlm/execution/openai/OpenAiVendorClient.java
@@ -22,15 +22,16 @@ import dev.promptlm.execution.util.PromptSpecUtils;
 import dev.promptlm.domain.promptspec.ChatCompletionRequest;
 import dev.promptlm.domain.promptspec.ChatCompletionResponse;
 import dev.promptlm.domain.promptspec.PromptSpec;
+import com.openai.models.ChatModel;
 import org.springframework.ai.chat.client.ChatClient;
 import org.springframework.ai.chat.model.ChatResponse;
 import org.springframework.ai.chat.prompt.Prompt;
 import org.springframework.ai.openai.OpenAiChatOptions;
 import org.springframework.ai.openai.OpenAiChatModel;
-import org.springframework.ai.openai.api.OpenAiApi;
 import org.springframework.stereotype.Component;
 
-import java.util.Arrays;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
 import java.util.Collections;
 import java.util.LinkedHashSet;
 import java.util.Locale;
@@ -40,7 +41,7 @@ import java.util.stream.Stream;
 
 /**
  * OpenAI vendor integration backed by Spring AI, supporting all models enumerated in
- * {@link OpenAiApi.ChatModel} plus a set of additional well-known aliases.
+ * {@link com.openai.models.ChatModel} plus a set of additional well-known aliases.
  */
 @Component
 public class OpenAiVendorClient implements SpringAiVendorClient {
@@ -76,25 +77,29 @@ public class OpenAiVendorClient implements SpringAiVendorClient {
     private static final Set<String> SUPPORTED_MODEL_ALIASES;
 
     static {
-        LinkedHashSet<String> catalog = Arrays.stream(OpenAiApi.ChatModel.values())
-                .map(OpenAiApi.ChatModel::getValue)
-                .map(String::trim)
-                .filter(value -> !value.isEmpty())
-                .collect(Collectors.toCollection(LinkedHashSet::new));
+        LinkedHashSet<String> catalog = new LinkedHashSet<>();
+        LinkedHashSet<String> aliases = new LinkedHashSet<>();
+        for (Field field : ChatModel.class.getDeclaredFields()) {
+            if (Modifier.isStatic(field.getModifiers())
+                    && ChatModel.class.isAssignableFrom(field.getType())) {
+                try {
+                    ChatModel model = (ChatModel) field.get(null);
+                    String modelId = model.asString();
+                    if (modelId != null && !modelId.isBlank()) {
+                        catalog.add(modelId.trim());
+                        aliases.add(normalizeModel(modelId));
+                        aliases.add(normalizeModel(field.getName()));
+                        aliases.add(normalizeModel(field.getName().replace('_', '-')));
+                    }
+                } catch (IllegalAccessException ignored) {
+                }
+            }
+        }
         catalog.addAll(ADDITIONAL_CATALOG_MODELS);
         CATALOG_MODELS = Collections.unmodifiableSet(catalog);
 
-        LinkedHashSet<String> aliases = Stream.concat(
-                        Arrays.stream(OpenAiApi.ChatModel.values())
-                                .flatMap(model -> Stream.of(
-                                        model.getValue(),
-                                        model.getName(),
-                                        model.name(),
-                                        model.name().replace('_', '-'))
-                                        .map(OpenAiVendorClient::normalizeModel)),
-                        ADDITIONAL_MODEL_ALIASES.stream())
-                .filter(alias -> !alias.isEmpty())
-                .collect(Collectors.toCollection(LinkedHashSet::new));
+        aliases.addAll(ADDITIONAL_MODEL_ALIASES);
+        aliases.removeIf(String::isEmpty);
         SUPPORTED_MODEL_ALIASES = Collections.unmodifiableSet(aliases);
     }
 

--- a/pom.xml
+++ b/pom.xml
@@ -19,7 +19,7 @@
         <maven-surefire-plugin.version>3.5.5</maven-surefire-plugin.version>
         <maven-jar-plugin.version>3.5.0</maven-jar-plugin.version>
         <spring-boot.version>4.0.6</spring-boot.version>
-        <spring-ai.version>2.0.0-M4</spring-ai.version>
+        <spring-ai.version>2.0.0-M5</spring-ai.version>
         <native.maven.plugin.version>0.11.5</native.maven.plugin.version>
         <spring-modulith.version>2.0.6</spring-modulith.version>
         <spring-boot-maven-plugin.version>4.0.6</spring-boot-maven-plugin.version>

